pkg/controller/gc/finalizer.go

```diff
@@ -0,0 +1,40 @@
+package gc
+
+import (
+	"github.com/atlassian/smith"
+
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	FinalizerDeleteResources string = smith.Domain + "/deleteResources"
+)
+
+func HasFinalizer(accessor meta_v1.Object, finalizer string) bool {
+	finalizers := accessor.GetFinalizers()
+	for _, f := range finalizers {
+		if f == finalizer {
+			return true
+		}
+	}
+	return false
+}
+
+func HasDeleteResourcesFinalizer(accessor meta_v1.Object) bool {
+	return HasFinalizer(accessor, FinalizerDeleteResources)
+}
+
+func AddDeleteResourcesFinalizer(accessor meta_v1.Object) {
+	accessor.SetFinalizers(append(accessor.GetFinalizers(), FinalizerDeleteResources))
```

Comment:
Should do the same as here probably https://github.com/atlassian/smith/blob/master/pkg/speccheck/speccheck.go#L101-L103
Don't need a set if there is a check for presence already, but sort them for consistency? I'm not sure if order matters at all. Hm...